app/lib/services/background_service.dart

```diff
@@ -88,12 +105,56 @@ Future<void> checkNodeStatus(String taskId) async {
           : '${nodesToNotify.length} Nodes Offline 🚨',
       body: bodyBuffer.toString().trim(),
       groupKey: 'offline_nodes',
+      type: NotificationType.nodeStatus,
+      additionalData: {
+        'nodeCount': nodesToNotify.length,
+        'nodeIds': nodesToNotify.map((n) => n.nodeId).toList(),
+      },
     );
   } catch (e) {
     logger.e('[BackgroundFetch] Error in checkNodeStatus for task $taskId: $e');
-  } finally {
-    logger.i('[BackgroundFetch] Finishing task $taskId');
-    BackgroundFetch.finish(taskId);
+  } 
+}
+
+Future<void> checkContractsAndNotify(
+    ProviderContainer container, String taskId) async {
+  try {
+    final List<ContractInfo> allContractsInGracePeriod = await container
+        .read(contractCheckServiceProvider)
+        .checkContractsState();
+
+    if (allContractsInGracePeriod.isNotEmpty) {
+      final bool contractNotificationsEnabled =
+          await isContractNotificationEnabled();
+      logger.i(
+          '[ContractsCheck] Contracts in grace period: ${allContractsInGracePeriod.length}. Contract Notifications enabled: $contractNotificationsEnabled');
+
+      if (contractNotificationsEnabled) {
```

Comment:
- I think we may include more info about contract to be user-friendly, like farm or node. what do u think ?

Reply:
Surely can be done but there's a [contract page](https://github.com/threefoldtech/threefold_connect/pull/1079) where users can check all their contracts details. Maybe we can refer to the contracts page but after its PR is merged?

Reply:
That's a good idae